### DISCUSSION TITLE
[Idea] Hummingbird: Optimization of css and js code loading only for specific pages

### DISCUSSION THREAD

**Discussion by Maofree** [PrestaShop/PrestaShop#34109] [Ideas]
Hi
would it be possible to load the css and js code in an optimized way? the css and js code relating to the single product page only on this page, that of the product list are on these pages, for the homepage, CMS, account pages and checkout the same thing. This way the performance would be higher and also more logical
it makes no sense for a visitor who is looking at a product page to also load the code related to the checkout, or accounts or cms pages into his browser ... same thing if he was looking at the categories or the checkout

**Comment by Maofree** [comment]
in practice when compiling several files are created and then use the directives present in theme.yml, using compression they would then be unified in a single file but it would have different weight for each page type

**Comment by Maofree** [comment]
what version of bootstrap is being used? the 4.7?

**Comment by Maofree** [comment]
you could take a cue from how it was done in the falcon theme

https://github.com/Oksydan/falcon/tree/master/config

**Comment by Hlavtox** [comment]
This is how it can worked in 1.6 and how it partially works now.

- If you include any assets (CSS and JS), they get included only on the pages you include them on. ✅
- CSS mechanism then makes bundles for each controller specifically. ✅
- Unfortunately, the main theme.css and theme.js files are common for all the controllers. 😞

**Comment by Maofree** [comment]
Hi @Hlavtox 
why is it not possible to use the same solution used in the falcon theme? try to have a look at the file assets.yml and stylesheets.tpl

https://github.com/Oksydan/falcon#register-assets

**Comment by matks** [comment]
Hi everyone in order to provide more visibility for the community I move this Discussion into PrestaShop/PrestaShop Discussions. The Hummingbird repository might not attract enough attention as suggested by @Oksydan
